hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferService.java

```diff
@@ -78,8 +88,24 @@
      */
     private NodeEngine nodeEngine;
 
+    private final QuorumService quorumService;
+
+    private final ConcurrentMap<String, Object> quorumConfigCache = new ConcurrentHashMap<String, Object>();
```

Comment:
can there be a leak in `quorumConfigCache` if i create and destroy a ring-buffer instance? who does the cleaning on ring-buffer destroy?

Reply:
Well, it's just a single config, not as many as there are instances, so it won't get duplicated if you create a new ringbuffer. It's an instance that already exists as in the Config class. So we should be fine without destroying

Reply:
Earlier I was wondering if this caching is compatible with dynamic config, wdyt @vbekiaris ?

Reply:
@tombujok from [here](https://github.com/hazelcast/hazelcast/pull/12024/files#diff-b78ce22c89cd149cf7ed6254f73ce559R334) i understand that it is populated for every new ring buffer instance. Maybe i am wrong?

Reply:
@tombujok with regards to dynamic config:
- new quorum names cannot be added on the fly (this is not supported anyway)
- a new ringbuffer (or other data structure) config referencing an existing quorum name can be added to the cluster via dynamic config
- a ringbuffer (or other data structure) config with a name for which configuration already exists will either be silently ignored (in case the new config is equal to the existing one) or fail with a `ConfigurationException` if new config is not equal to the existing config.

Reply:
@vbekiaris ok that sounds good in this case.

@ahmetmircik it's just a mapping data-structure name -> quorumName, if it exists in the mapping it won't be added.

Reply:
@ahmetmircik but you are right, if your only intent is to create maps with random name, run one operation, and destroy maps, then this cache will exhaust all memory at some point...